merlin/systems/triton/models/executor_model.py

```diff
@@ -0,0 +1,145 @@
+        # model_repository argument became path to model.py
+        if str(repository_path).endswith(".py"):
+            repository_path = repository_path.parent.parent
+
+        ensemble_path = repository_path / str(args["model_version"]) / "ensemble"
+
+        self.executor = LocalExecutor()
+        self.ensemble = Ensemble.load(str(ensemble_path))
+
+        for node in list(postorder_iter_nodes(self.ensemble.graph.output_node)):
+            if hasattr(node.op, "load_artifacts"):
+                node.op.load_artifacts(str(ensemble_path))
+
+    def execute(self, requests):
+        """Receives a list of pb_utils.InferenceRequest as the only argument. This
+        function is called when an inference is requested for this model. Depending on the
+        batching configuration (e.g. Dynamic Batching) used, `requests` may contain
+        multiple requests. Every Python model, must create one pb_utils.InferenceResponse
+        for every pb_utils.InferenceRequest in `requests`. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse.
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        list
+          A list of pb_utils.InferenceResponse. The length of this list must
+          be the same as `requests`
+        """
+        responses = []
+
+        for request in requests:
+            try:
+                # transform the triton tensors to a dict of name:numpy tensor
+                input_tensors = {
+                    name: pb_utils.get_input_tensor_by_name(request, name).as_numpy()
+                    for name in self.ensemble.input_schema.column_names
+                }
+
+                return_values = self.executor.transform(
+                    DictArray(input_tensors),
+                    [self.ensemble.graph.output_node],
+                )
+
+                output_tensors = []
+                for name, data in return_values.items():
+                    if isinstance(data, pb_utils.Tensor):
+                        output_tensors.append(data)
+                        continue
+                    data = data.get() if hasattr(data, "get") else data
+                    tensor = pb_utils.Tensor(name, data)
+                    output_tensors.append(tensor)
+
+                responses.append(pb_utils.InferenceResponse(output_tensors))
+
+            except Exception:  # pylint: disable=broad-except
+                exc_type, exc_value, exc_traceback = sys.exc_info()
+                tb_string = repr(traceback.extract_tb(exc_traceback))
+                responses.append(
+                    pb_utils.InferenceResponse(
+                        tensors=[], error=f"{exc_type}, {exc_value}, {tb_string}"
+                    )
+                )
+
+        return responses
+
+
+def _parse_model_repository(model_repository: str) -> str:
```

Comment:
Looks like this was intended to be used in `TritonPythonModel`?

Reply:
Good catch!